cassandra-executor/src/main/java/io/mesosphere/mesos/frameworks/cassandra/CassandraExecutor.java

```diff
@@ -168,9 +188,59 @@ private Process launchCassandraNodeTask(@NotNull final Marker taskIdMarker, @Not
 
     @NotNull
     private CassandraNodeHealthCheckDetails performHealthCheck(@NotNull final CassandraNodeHealthCheckTask healthCheckTask) {
-        return CassandraNodeHealthCheckDetails.newBuilder()
-            .setHealthy(true)
-            .build();
+        CassandraNodeHealthCheckDetails.Builder builder = CassandraNodeHealthCheckDetails.newBuilder();
+        try (JmxConnect jmx = new JmxConnect(getHostAddress(), (int) healthCheckTask.getJmxPort() & 0xffff)) {
+
+            // TODO Robert : we should add some "timeout" that allows C* to start and join
+            // I.e. track status/operation-mode changes
+
+            CassandraNodeInfo info = buildInfo(jmx);
+            builder.setHealthy(true)
+                    .setInfo(info);
+            LOGGER.info("Healthcheck succeeded: operationMode:{} joined:{} gossip:{} native:{} rpc:{} uptime:{}s endpoint:{}, dc:{}, rack:{}, hostId:{}, version:{}",
+                    info.getOperationMode(),
+                    info.getJoined(),
+                    info.getGossipRunning(),
+                    info.getNativeTransportRunning(),
+                    info.getRpcServerRunning(),
+                    info.getUptimeSeconds(),
+                    info.getEndpoint(),
+                    info.getDataCenter(),
+                    info.getRack(),
+                    info.getHostId(),
+                    info.getVersion());
+        } catch (Exception e) {
+            LOGGER.warn("Healthcheck failed.", e);
+            builder.setMsg(e.toString());
+        }
+        return builder.build();
+    }
+
+    private CassandraNodeInfo buildInfo(JmxConnect jmx) throws UnknownHostException {
+        Nodetool nodetool = new Nodetool(jmx);
+
+        String endpoint = nodetool.getEndpoint();
+
+        // TODO HC response might be a bit over-engineered
+
+        // C* should be considered healthy, if the information can be collected.
+        // All flags can be manually set by any administrator and represent a valid state.
+        return CassandraNodeInfo.newBuilder()
+                .setOperationMode(nodetool.getOperationMode())
+                .setJoined(nodetool.isJoined())
+                .setGoosipInitialized(nodetool.isGossipInitialized())
+                .setGossipRunning(nodetool.isGossipRunning())
+                .setNativeTransportRunning(nodetool.isNativeTransportRunning())
+                .setRpcServerRunning(nodetool.isRPCServerRunning())
+                .setUptimeSeconds(nodetool.getUptimeInMillis() / 1000L)
```

Comment:
was intentionally to make it more "human readable" - but using the millis is more accurate. changed to millis.